merger.go

```diff
@@ -397,32 +397,25 @@ func (m *Merger) launch() (err error) {
 }
 
 func (m *Merger) retrieveListOfFiles(ctx context.Context) (tooOld []string, seenInCache []string, good []string, err error) {
-	var count int
+
+	canonicalGoodFiles := make(map[string]struct{})
```

Comment:
Why a `struct{}` type and not a `bool`?
```suggestion
	canonicalGoodFiles := make(map[string]bool)
```

Reply:
it is the smallest footprint in memory, an empty struct is the recommended way for a map where we just want the key